### ISSUE TITLE
Add a site-wide dark mode

### ISSUE THREAD

**Issue by lemystic** [scratchfoundation/scratch-www#7333]
This is simply a feature request to add dark mode to places _outside of the editor_ through the device theme or as a toggle in the navbar.

Potential dark mode made with edits made to `src/_colors.scss`:

<img src=https://assets.scratch.mit.edu/66c1717644b743501eed586060fffba6.png height=350>

People on the discussion forums have been asking for this feature ([link](https://scratch.mit.edu/discuss/topic/217797/)) for many years and the topic is going strong at 130+ pages. Many modern sites have a dark mode, and much of the Scratch community would appreciate the feature.

While you can use an extension or userscript, discussion of these things in the forums are [banned](https://scratch.mit.edu/discuss/post/2907564/) and a native solution is still preferred by most. 

I know a dark mode is a simple aesthetic change which could have whatever priority, but this would finally end so many requests from the users and everyone would be satisfied.

This should only be added to finished 3.0 pages for consistency, and can be introduced to newly migrated pages. The editor already has a feature request [here](https://github.com/LLK/scratch-gui/issues/59), but this is for the whole site.

**Comment by tonyzhigit** [comment]
Maybe create a web page to let the users customize the page's colour?

**Comment by themysticsavages** [comment]
> Maybe create a web page to let the users customize the page's colour?

I don't know, doesn't it sound a little excessive for Scratch? It would also require a lot of modification to the CSS to work consistently across pages. Thanks, but dark mode should be good for now. 

The requests for dark mode are certainly higher than custom theming at the moment.

**Comment by showierdata9978** [comment]
I agree dark mode needs to be added before ustom theming

**Comment by veloce-vasta** [comment]
Has the ST not noticed the threads before this was made??

**Comment by themysticsavages** [comment]
> Has the ST not noticed the threads before this was made??

Well, we would have some kind of status regarding dark mode at this point, so they probably just look it over.

**Comment by Secret-chest** [comment]
Wouldn't custom themeing go with this? Adding dark mode implies making the CSS use custom properties

**Comment by showierdata9978** [comment]
> Wouldn't custom themeing go with this? Adding dark mode implies making the CSS use custom properties

no, not really. all dark mode wold do is change the colors to be more dark...

**Comment by showierdata9978** [comment]
> > > Wouldn't custom themeing go with this? Adding dark mode implies making the CSS use custom properties
> > 
> > 
> > no, not really. all dark mode wold do is change the colors to be more dark...
> 
> Yes, and by doing that you change the colours in the CSS so why not add custom colours.

because its easier to implement, and only a single extra button is needed to do it lmao

**Comment by showierdata9978** [comment]
Alot more logic is required for EVERY SINGLE SCRATCH USER

Dark mode is a single if statement, and a single cookie.

**Comment by themysticsavages** [comment]
The difficulty of changing the site colors cannot be understated. It takes a minute to build it in development, thanks to Webpack (should really switch to Vite LOL), and all the color changes you make have to be WCAG-compliant. It really is kind of a pain

**Comment by lemystic** [comment]
Bump

**Comment by 659in** [comment]
I see this as a win (I am 56-s)

**Comment by teamsonic2011** [comment]
Support, light mode hurts my eyes.
Wait how does GitHub work

**Comment by medlans** [comment]
Another mockup:

https://github.com/scratchfoundation/scratch-gui/issues/59#issuecomment-2537587844
